capture/src/disk.cpp

```diff
@@ -53,10 +59,30 @@ void write_to_disk(
 
         if (disk_write_enabled)
         {
+            // Check free disk space (but not every single frame)
+            if (frame_count % 100 == 0)
+            {
+                int rtn = statvfs("/", &disk_stats);
+                if (rtn != 0)
+                {
+                    warnx("Tried to check disk space with statvfs but got return code %d.", rtn);
```

Comment:
I like the use of the `<err.h>` functions! Though in this case, you'll actually want to do something more like this, since the return value of `statvfs` on failure is simply `-1` and the actual info is conveyed via `errno`:
```cpp
warn("Tried to check disk space with statvfs but the call failed");
```
And that will print out `"Tried to check disk space with statvfs but the call failed: No such file or directory"`.